extensions/llamacpp-extension/src/index.ts

```diff
@@ -1576,6 +1576,16 @@ export default class llamacpp_extension extends AIEngine {
     }
   }
 
+  async getRecommendedBackend(): Promise<string> {
+    try {
+      const supportedBackends = await listSupportedBackends()
+      return this.determineBestBackend(supportedBackends)
+    } catch (error) {
+      logger.error('Failed to get recommended backend:', error)
+      return ''
+    }
```

Comment:
Since this function returns a promise, there is no need to try catch in place
```suggestion
    return listSupportedBackends().then(this.determineBestBackend)
```